src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -250,9 +247,9 @@ void Gather::prepareParams() {
         specIdxAndAfterAxSizeB = specIndicesSize * afterAxisSizeInBytes;
         totalWork = beforeBatchSize * betweenBatchAndAxisSize * specIndicesSize * afterAxisSize;
     }
-
+    createOrUpdateJitKernelIfNeeded();
     const auto& selectedPD = getSelectedPrimitiveDescriptor();
-    if (jitKernel && jitKernel->isSupportedConfiguration(afterAxisSize)) {
+    if (jitKernel) {
```

Comment:
If the Kernel supports all cases, the selectedPD should be set just once in fn *initSupportedPrimitiveDescriptors*. In the previous implementation execution could fallback on reference implementation depend on Axis value thus the selectedPD need to be changed.

Reply:
This PR is about refactoring only and the kernel doesn't support all cases here. The `jitKernel == nullptr` for unsupported cases. I will prepare the improved kernel which supports all cases in separate PR.